tasks: [
    {
      id: 0,
      name: "Casino_Gate",
      plane: 0,
      search_depth: 1,
      search_accel: 0.8,
      pass_thru_duration: 6.0, # [s]
      gate_zcenter_offset: 0.25, # fraction of frame height 
      gate_width: 0.6, # fraction of frame width
      objects: ["Casino_Gate_Black"],
      thresholds: [0.5],
      align_thresh: 50, # [pixels]
      bbox_thresh: 50, # [pixels]
      detection_duration: 1, # [s]
      detections_req: 5 # [number]
    },
    {
      id: 1,
      name: "Path_Marker1",
      plane: 1,
      search_depth: 1,
      search_accel: 0.8,
      path_angle: 135,
      objects: ["Path_Marker"],
      thresholds: [0.3],
      align_thresh: 30, # [pixels]
      bbox_thresh: 30, # [pixels]
      detection_duration: 1, # [s]
      detections_req: 8 # [number]
    },
    {
      id: 2,
      name: "Dice",
      plane: 0,
      search_depth: 3,
      search_accel: 0.8,
      bump_duration: 4, # [s]
      backup_duration: 6, # [s]
      move_over_duration: 1, # [s]
      dice_bbox_width: 0.25, # fraction of frame width
      upper_dice_zcenter_offset: 0.25, # fraction of frame height
      objects: ["Dice1", "Dice2", "Dice5", "Dice6"],
      preferred_objects: [6, 5], # Just use actual dice block number
      thresholds: [0.6, 0.6, 0.6, 0.6],
      align_thresh: 30, # [pixels]
      bbox_thresh: 30, # [pixels]
      detection_duration: 1, # [s]
      detections_req: 8 # [number]
    },
    {
      id: 3,
      name: "Path_Marker2",
      plane: 1,
      search_depth: 2,
      search_accel: 0.8,
      path_angle: 135,
      objects: ["Path_Marker"],
      thresholds: [0.6],
      align_thresh: 30, # [pixels]
      bbox_thresh: 30, # [pixels]
      detection_duration: 1, # [s]
      detections_req: 8 # [number]
    },
    {
      id: 4,
      name: "Slots",
      plane: 0,
      search_depth: .5,
      search_accel: 0.8,
      objects: ["Slots_Fruit", "Slots_Big_Red"],
      torpedo_offset: {
        port: { y: 100, z: 0 },
        stbd: { y: -100, z: 0 }
      },
      big_red_bbox_height: 0.5,
      fruit_bbox_height: 0.5,
      pneumatics_duration: 200,
      thresholds: [0.1, 0.1],
      align_thresh: 30, # [pixels]
      bbox_thresh: 30, # [pixels]
      detection_duration: 1, # [s]
      detections_req: 8 # [number]
    },
    {
      id: 5,
      name: "Buy_Gold_Chip1",
      plane: 0,
      search_depth: .5,
      search_accel: 0.8,
      burn_time: 2.5,
      back_off_time: 2.5,
      bbox_height: 0.3, # fraction of frame height
      objects: ["Buy_Gold_Chip"],
      thresholds: [0.1],
      align_thresh: 30, # [pixels]
      bbox_thresh: 30, # [pixels]
      detection_duration: 1, # [s]
      detections_req: 8 # [number]
    },
    {
      id: 6,
      name: "Roulette",
      plane: 1,
      search_depth: 2.0,
      search_accel: 0.8,
      roulette_bbox_height: 0.7, # fraction of frame height
      roulette_ycenter_offset: 0.2, # fraction of frame width
      objects: ["Roulette"],
      thresholds: [0.1],
      drop_duration: 4.0,
      align_thresh: 30, # [pixels]
      bbox_thresh: 30, # [pixels]
      detection_duration: 1.0, # [s]
      detections_req: 8 # [number]
    },
    {
      id: 7,
      name: "Buy_Gold_Chip2",
      plane: 0,
      search_depth: 3,
      search_accel: 0.8,
      burn_time: 2.5,
      back_off_time: 2.5,
      objects: ["Buy_Gold_Chip"],
      thresholds: [0.5],
      align_thresh: 30, # [pixels]
      bbox_thresh: 30, # [pixels]
      detection_duration: 1, # [s]
      detections_req: 8 # [number]
    },
    {
      id: 8,
      name: "Cash_In",
      plane: 0,
      search_depth: 2,
      search_accel: 0.8,
      objects: ["Cash_In"],
      thresholds: [0.5],
      align_thresh: 30, # [pixels]
      bbox_thresh: 30, # [pixels]
      detection_duration: 1, # [s]
      detections_req: 8 # [number]
    }
  ]
